src/chain/CacheManager.sol

```diff
@@ -94,7 +98,64 @@ contract CacheManager {
         }
     }
 
-    /// Sends all revenue to the network fee account.
+    /// @notice Returns all entries in the cache. Might revert if the cache is too large.
+    function getEntries() external view returns (Entry[] memory) {
+        return entries;
+    }
+
+    /// @notice Returns the `k` smallest entries in the cache sorted in ascending order.
+    ///         If the cache have less than `k` entries, returns all entries.
+    function getSmallestEntries(uint256 k) public view returns (Entry[] memory result) {
+        if (bids.length() < k) {
+            k = bids.length();
+        }
+        result = new Entry[](k);
+        uint256[] memory kbids = bids.smallest(k);
+        for (uint256 i = 0; i < k; i++) {
+            (, uint64 index) = _getBid(kbids[i]);
+            result[i] = entries[index];
+        }
+    }
+
+    /// @notice Returns the minimum bid required to cache a program of the given size.
+    ///         Value returned here is the minimum bid that you can send with msg.value
+    function getMinBid(uint64 size) public view returns (uint192 min) {
+        if (size > cacheSize) {
+            revert AsmTooLarge(size, 0, cacheSize);
+        }
+
+        size = size >= MIN_CODESIZE ? size : MIN_CODESIZE;
+        uint256 totalSize = queueSize + size;
+        if (totalSize <= cacheSize) {
+            return 0;
+        }
+        uint256 needToFree = totalSize - cacheSize;
+
+        // code size is at least 4Kb, and vary no more than 10x right now, so we can safely assume
```

Comment:
4Kb isn't the minimum, it's just a minimum we use in this contract for defensive programming purposes. The range is still only an order of magnitude though currently